### DISCUSSION TITLE
BeamRunPythonPipelineOperator labels are not added if defined in default_pipeline_options

### DISCUSSION THREAD

**Discussion by norcs** [apache/airflow#20682] [General]
I am using the [BeamRunPythonPipelineOperator](https://github.com/apache/airflow/blob/e8b5ab9efe93f826a2b521b5d1cac0404354c3b4/airflow/providers/apache/beam/operators/beam.py#L102) and I tried to add my labels to a dataflow job using the `default_pipeline_options` parameter but the labels are not showing up in the Job Info on GCP. I tried the same with BeamRunJavaPipelineOperator and the labels were added to the job.
From the documentation:
_Note that both default_pipeline_options and pipeline_options will be merged to specify pipeline execution parameter, and default_pipeline_options is expected to save high-level pipeline_options, for instances, project and zone information, which apply to all Apache Beam operators in the DAG._

In the `BeamRunPythonPipelineOperator` `__init__` default labels are set
```
...
        self.default_pipeline_options = default_pipeline_options or {}
        self.pipeline_options = pipeline_options or {}
        self.pipeline_options.setdefault("labels", {}).update(
            {"airflow-version": "v" + version.replace(".", "-").replace("+", "-")}
        )
...
```

and in
```
    def execute(self, context: 'Context'):
        """Execute the Apache Beam Pipeline."""
        self.beam_hook = BeamHook(runner=self.runner)
        pipeline_options = self.default_pipeline_options.copy()
        process_line_callback: Optional[Callable] = None
        is_dataflow = self.runner.lower() == BeamRunnerType.DataflowRunner.lower()
        dataflow_job_name: Optional[str] = None

        if is_dataflow:
            dataflow_job_name, pipeline_options, process_line_callback = self._set_dataflow(
                pipeline_options=pipeline_options, job_name_variable_key="job_name"
            )

        pipeline_options.update(self.pipeline_options)
...
```

So basically the labels in `default_pipeline_options` are overwritten with the labels set to `pipeline_options` in the `__init__` (`{"airflow-version": "v" + version.replace(".", "-").replace("+", "-")}`). I also found that in `self._set_dataflow(...)` the labels are set again. I checked the `__init__` in `BeamRunJavaPipelineOperator` and the labels are not set there.

So my questions are why the different behaviour in the python and java version of this operator and what is the reason behind it.
